src/java/org/apache/cassandra/utils/concurrent/LoadingMap.java

```diff
@@ -52,14 +56,24 @@
     // It also ensures exactly-once semantics for the update operation.
     private final Map<K, CompletableFuture<V>> internalMap;
 
+    private final Duration timeout;
+
     public LoadingMap()
     {
-        this.internalMap = new NonBlockingHashMap<>();
+        this(8);
     }
 
     public LoadingMap(int initialSize)
     {
+        this(initialSize, Duration.ZERO);
+    }
+
+    public LoadingMap(int initialSize, Duration timeout)
```

Comment:
`timeout` becomes a part of the API now. 
We need clear javadocs that `Duration.ZERO` means unlimited timeout, and that `(int)` ctor creates a map without timeout etc